### PR TITLE
new package: libbullet

### PR DESCRIPTION
**Pull request by pvonmoradi**
Adds [bulletphysics](https://github.com/bulletphysics/bullet3) library

### CODE REVIEW

**Review by 2096779623 — CHANGES_REQUESTED** [review]
* `TERMUX_PKG_HOMEPAGE: NON-HTTPS (acceptable)`
It will redirect to `https://pybullet.org/wordpress/` From readme.md of GitHub homepage, change `TERMUX_PKG_HOMEPAGE` to`https://kubric.readthedocs.io`

* `TERMUX_PKG_DESCRIPTION: TOO LONG (allowed: 100 characters max)`

**Comment by pvonmoradi** [comment]
> * `TERMUX_PKG_HOMEPAGE: NON-HTTPS (acceptable)`
>       It will redirect to `https://pybullet.org/wordpress/` From readme.md of GitHub homepage, change `TERMUX_PKG_HOMEPAGE` to`https://kubric.readthedocs.io`
> 
>     * `TERMUX_PKG_DESCRIPTION: TOO LONG (allowed: 100 characters max)`

kubric is a something that interfaces with PyBullet. I'll link to the github repo.

File: packages/libbullet/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/bulletphysics/bullet3"
+TERMUX_PKG_DESCRIPTION="SKD for real-time collision detection and multi-physics simulation for robotics, VR, etc"
```
Comment (truboxl) [inline packages/libbullet/build.sh:2]: Typo

File: packages/libbullet/build.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+TERMUX_PKG_GROUPS="science"
+TERMUX_PKG_LICENSE="ZLIB"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="3.24"
+TERMUX_PKG_SRCURL="https://github.com/bulletphysics/bullet3/archive/refs/tags/$TERMUX_PKG_VERSION.tar.gz"
+TERMUX_PKG_SHA256=6b1e987d6f8156fa8a6468652f4eaad17b3e11252c9870359e5bca693e35780b
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_FORCE_CMAKE=true
```
Comment (truboxl) [inline packages/libbullet/build.sh:10]: Probably not needed. Also please add `TERMUX_PKG_DEPENDS="libc++"`

File: packages/libbullet/build.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+-DUSE_DOUBLE_PRECISION=OFF
+-DBUILD_PYBULLET=OFF
+-DBUILD_BULLET2_DEMOS=OFF
+-DBUILD_CPU_DEMOS=OFF
+-DUSE_GLUT=OFF
+-DBT_USE_EGL=OFF
+-DUSE_GRAPHICAL_BENCHMARK=OFF
+-DBUILD_EXTRAS=OFF
+-DBUILD_UNIT_TESTS=OFF
+-DCMAKE_BUILD_TYPE=Release
```
Comment (truboxl) [inline packages/libbullet/build.sh:24]: This was already set in https://github.com/termux/termux-packages/blob/master/scripts/build/configure/termux_step_configure_cmake.sh#L35

**Review by truboxl — CHANGES_REQUESTED** [review]

File: packages/libbullet/build.sh (hunk 2 above)
Comment (pvonmoradi) [reply packages/libbullet/build.sh]: All C or C++ libs should have a dependency on libc++?

Comment (truboxl) [reply packages/libbullet/build.sh]: C++ usually will depend on libc++_shared.so, can check any lib*.so file or executable using ldd or readelf or objdump

**Review by truboxl — APPROVED** [review]

**Comment by Grimler91** [comment]
Thanks!

**Comment by robertkirkman** [comment]
@pvonmoradi someone needs this with `-DUSE_DOUBLE_PRECISION=ON` in order to compile this https://gitlab.com/OpenMW/openmw . however, you added this with `-DUSE_DOUBLE_PRECISION=OFF`. is there a particular reason you needed it off, or would it be ok to turn it on?

**Comment by pvonmoradi** [comment]
@robertkirkman From what I remember, I did it for performance reasons. There are 4 targets for termux packages, namely, aarch64, arm, i686 and x86_64. Two of these are 32-bit platforms. I have not checked if setting `-DUSE_DOUBLE_PRECISION=ON` would generate a working binary on 32 bit devices or how bad the performance hit of double float calculations on a 32-bit SoC would be.
Probably the best solution here would be to modify the script to use double on 64 bit systems. I don't have termux-packages dev stuff on my machine set up now. My guess is even with ON it would build fine.

**Comment by robertkirkman** [comment]
Thanks for replying, that's actually a very important concern. I will make sure to test what happens with this use case on 32-bit devices before considering applying it to them.

**Pull request merged**